### PR TITLE
UN-574: Remove unused `CollectionHighlights` and `ResultsPage`

### PR DESCRIPTION
**Pull request by jamesbiggs**
Ticket URL: https://national-archives.atlassian.net/browse/UN-574

## About these changes

Removed all code relating to `CollectionHighlights` and `ResultsPage`

## How to check these changes

Where possible, provide guidance to help your reviewer

## Before assigning to reviewer, please make sure you have

- [x] Checked things thoroughly before handing over to reviewer.
- [x] Checked PR title starts with ticket number as per project conventions to help us keep track of changes.
- [x] Ensured that PR includes only commits relevant to the ticket.
- [x] Waited for all CI jobs to pass before requesting a review.
- [x] Added/updated tests and documentation where relevant.

## Merging PR guidance

Follow [docs\developer-guide\contributing.md](https://nationalarchives.github.io/ds-wagtail/developer-guide/contributing/)

## Deployment guidance

Follow [docs\infra\environments.md](https://nationalarchives.github.io/ds-wagtail/infra/environments/)

### CODE REVIEW

File: etna/search/views.py [hunk 1]
```diff
@@ -670,36 +669,6 @@ def add_article_page_for_url(self, page: Page) -> None:
                         f"WebsiteSearchView:scraped/ingested url={absolute_url} not found in wagtail_pages={wagtail_pages}"
                     )
 
-    def add_results_page_for_url(self, page: Page) -> None:
```
Comment (ababic) [inline etna/search/views.py:673]: @jamesbiggs We also need to make sure any code calling this method is updated :)

**Comment by ababic** [comment]
Couple of small things @jamesbiggs:
1. Remove the `collections/results_page.html` template too
2. Are there any styles created for that page/template that can be removed  too?

**Comment by jamesbiggs** [comment]
Hey @ababic, I've made the changes you requested :)

**Review by ababic — APPROVED** [review]
It sure does feel nice to shed this code! Thanks @jamesbiggs!

**Pull request merged**